commands/plugin_e2e_test.go

```diff
@@ -0,0 +1,147 @@
+package commands
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/spf13/cobra"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestPluginEnableCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	pluginID := "com.mattermost.demo-plugin"
+	pluginURL := filepath.Join(os.Getenv("MM_SERVER_PATH"), "tests", "testplugin.tar.gz")
+	nonExistentPluginID := "nonExistentPluginID"
+
+	installPlugin(s, pluginID, pluginURL)
+	defer removePlugin(s, pluginID)
+
+	s.RunForSystemAdminAndLocal("Successful enable plugin", func(c client.Client) {
+		printer.Clean()
+
+		appErr := s.th.App.DisablePlugin(pluginID)
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+
+		cmd := &cobra.Command{}
+		err := pluginEnableCmdF(c, cmd, []string{pluginID})
+		s.Require().Nil(err)
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 1)
+		s.Require().Len(plugins.Inactive, 0)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0], "Enabled plugin: "+pluginID)
+
+	})
+
+	s.Run("error for enable plugin", func() {
+		printer.Clean()
+
+		appErr := s.th.App.DisablePlugin(pluginID)
+		s.Require().Nil(appErr)
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+
+		cmd := &cobra.Command{}
+		_ = pluginEnableCmdF(s.th.Client, cmd, []string{pluginID})
+		//err = pluginEnableCmdF(c, cmd, []string{pluginID})
+		//s.Require().Error(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to enable plugin: "+pluginID+". Error: : You do not have the appropriate permissions., ")
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+	})
+
+	s.RunForSystemAdminAndLocal("error for enabling non existent plugin", func(c client.Client) {
+		printer.Clean()
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+
+		cmd := &cobra.Command{}
+		_ = pluginEnableCmdF(c, cmd, []string{nonExistentPluginID})
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to enable plugin: "+nonExistentPluginID+". Error: : Plugin is not installed., ")
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+	})
+
+	s.Run("error for enabling non existent plugin", func() {
```

Comment:
You can remove this case. As it is not required that we already test for regular users don't have permission and also if a non-existent plugin-id could be enabled.